build.gradle

```diff
@@ -5,7 +5,8 @@ plugins {
     id 'net.nemerosa.versioning' version '3.0.0'
     id 'com.github.ben-manes.versions' version '0.44.0'
     id 'io.codearte.nexus-staging' version '0.30.0'
-    id 'org.openjfx.javafxplugin' version '0.0.13'
+    id 'org.openjfx.javafxplugin' version '0.1.0' apply false
+    id 'org.javamodularity.moduleplugin' version '1.8.12' apply false
```

Comment:
Why do we still need `org.javamodularity.moduleplugin`?

Reply:
If it is because automatic modules are not recognised with gradle 8.3, we might consider `org.gradlex:extra-java-module-info:1.4.1` instead?

Reply:
It was transitively depending on it via the javafx plugin previously

Reply:
We can switch to using `org.gradlex:extra-java-module-info:1.4.2` if desired, although I haven't used that plugin before.

I believe these would be the minimal amount of changes (explicitly declaring all legacy jars as automatic modules)

```gradle
// build.gradle
// We can remove modularity plugin
java {
    modularity.inferModulePath.set(true)
}
...
// FXyz-Samples/build.gradle
extraJavaModuleInfo {
    automaticModule("eu.mihosoft.vrl.jcsg:jcsg", "jcsg")
    automaticModule("eu.mihosoft.vvecmath:vvecmath", "vvecmath")
    automaticModule("org.orbisgis:poly2tri-core", "poly2tri.core")
    automaticModule("org.slf4j:slf4j-api", "slf4j.api")
    automaticModule("org.reactfx:reactfx", "reactfx")
    automaticModule("org.fxmisc.easybind:easybind", "easybind")
}
...
// FXyz-Client/build.gradle
extraJavaModuleInfo {
    automaticModule("org.reactfx:reactfx", "reactfx")
}
...
// FXyz-Core/build.gradle
extraJavaModuleInfo {
    automaticModule("eu.mihosoft.vrl.jcsg:jcsg", "jcsg")
    automaticModule("eu.mihosoft.vvecmath:vvecmath", "vvecmath")
    automaticModule("org.orbisgis:poly2tri-core", "poly2tri.core")
    automaticModule("org.slf4j:slf4j-api", "slf4j.api")
}
```

Reply:
The gradlex plugin is from @jjohannes, so I guess this is the way to go, in line with the changes in the JavaFX gradle plugin.

Reply:
Some notes:
- Automatic-Modules are recognized by Gradle, if there is an "Automatic-Module-Name" entry in the Manifest. So I am not sure if you would need all the things listed above. E.g. slf4j is a "real" module in the latest version. If you run the build with the plugin without defining any patching, it will tell you which Jars have to be patched.
- If you use the plugin and do patching in a library (not an application) build like this one, you should be carful which Module Names you choose if the names also turn up int your `module-info.java`. Because if you choose a different one as the one that Java would give based on the Jar name, users of the library would have to do the same patching. For example, in the list about "slf4j.api" could be problematic: It's true if the Jar has no automatic module name in the version you are using, but the official name in the current version is "org.slf4j" (without api). 

It is still a very messy situation right now, especially for libraries that have transitive dependencies which are not yet completely modularized. :( I think I would do this in an separate PR and continue to use the other plugin for now if that is what was done implicitly before.
Happy to give more pointers or do a review if you do the change in another PR.

Reply:
Thanks great point on slf4j naming, I didn't even think about it. I'm hopeful we can remove the slf4j entries when we merge https://github.com/FXyz/FXyz/pull/173 which upgrades it to 2.x.
All the ones added above were ones that the plugin errored on via `Not a module and no mapping defined: xyz.jar`

I also just opened https://github.com/openjfx/javafx-gradle-plugin/pull/154 to update docs on the removal of `org.javamodularity.moduleplugin` for others that might encounter this situation.

Reply:
#173 merged.
@samypr100 Do you want to update this PR? Do we merge this one before #172?

Reply:
I think we'll want to merge this one last of the 3

Reply:
We can switch to use the org.gradlex:extra-java-module-info plugin if you're alright with that or I'd be okay doing that on a different PR. Whichever you prefer.

Reply:
Ok, let's go for now with your current approach, and we'll switch plugins if needed.